### DISCUSSION TITLE
Unable to Load COM object DLL

### DISCUSSION THREAD

**Discussion by GopinathRamajayam** [pythonnet/pythonnet#1739] [Q&A]
### Environment

-   Pythonnet version: 2.5.2
-   Python version: 3.9.9 (64 Bit)
-   Operating System: Windows 64 Bit
-   .NET Runtime: V4.0

### Details

I am trying to load DLL (add a reference to COM objects) from Simatic Step 7 Manager.
 
From the documentation :
In order to use the command interface, you must make two entries for the project. To do so, select in Visual Studio the menu command Project > Add Reference > COM.  You can find the accompanying DLLs under STEP7/S7bin/s7abatcx.dll and STEP7/S7bin/s7hcom_x.dll. 

#### My Code :
import sys
visualStudio_dll = r'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\PublicAssemblies\envdte.dll'
clr.AddReference(str(visualStudio_dll))
from EnvDTE import DTE

S7ABATCX_dll = r'C:\Program Files (x86)\Siemens\Step7\S7BIN\S7ABATCX.DLL'
S7hcom_x_dll = r'C:\Program Files (x86)\Siemens\Step7\S7BIN\S7hcom_x.dll'
clr.AddReference(str(S7ABATCX_dll))
clr.AddReference(str(S7hcom_x_dll))

### Things I tried 
https://stackoverflow.com/questions/13259617/python-for-net-unable-to-find-assembly-error

I tried a few things from this StackOverflow post. Unfortunately, it didn't work out.

There is no Block for the DLLs.

I tried without .dll in the path 
    S7ABATCX_dll = r'C:\Program Files (x86)\Siemens\Step7\S7BIN\S7ABATCX'
    clr.AddReference(str(S7ABATCX_dll))

I tried adding the sys path for the dll.
    import sys
    sys.path.append(os.path.dirname(S7ABATCX_dll))
    sys.path.append(r"C:/Program Files (x86)/Siemens/Step7/S7BIN")

### Works in C# 
I am able to add this COM (S7ABATCX.DLL and S7hcom_x.dll) reference in C# Visual Studio 2019. But I am not able to do it in Python using PythonNet.

### Traceback
The error message I am getting, 
'''
Traceback (most recent call last):
  File "C:\Python39\lib\code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 3, in <module>
System.IO.FileNotFoundException: Unable to find assembly 'C:\Program Files (x86)\Siemens\Step7\S7BIN\S7ABATCX.DLL'.
   at Python.Runtime.CLRModule.AddReference(String name)
'''

**Comment by GopinathRamajayam** [comment]
Added .Net version

**Comment by kazu8-narita** [comment]
I worried you trying to load a 32bit dll with 64bit python, you will need 32bit python?(or 64bit dll?)

**Comment by GopinathRamajayam** [comment]
Thank you for your input, Kazu Narita. 

The DLLs are 32-bit. So, I have tried using 32bit python and Pythonnet. It still throws the same error. 
I have used the COMTypes python package. It works for one of the dll (S7ABATCX). I am still figuring out the other dll (S7hcom_x).
